authentik/sources/ldap/models.py

```diff
@@ -99,6 +99,11 @@ class LDAPSource(Source):
         help_text=_("Property mappings used for group creation/updating."),
     )
 
+    password_login_update_internal_password = models.BooleanField(
+        default=True,
```

Comment:
Personally I would advocate for having this default to `False` as having this active implies some non-obvious side effects and pitfalls. This should be a conscious opt-in with careful considerations and only once people experience performance bottlenecks without it.

Implications of having this enabled:
- Updating the LDAP password does not invalidate the local password, this means that old password are still valid indefinitely
- Logging in via LDAP credentials overwrites the local password if users have different pwds for both
- The security measures made to secure the passwords (salting, peppering, hash/PBKDF function) in Authentik might be different from the LDAP backend. Users might not be aware of this...

Reply:
I understand your point, but on the other hand that would mean breaking a behavior that existed for 4 years

Reply:
Then the migration should be configured such that existing sources keep the existing behavior but new ones will have this disabled

Reply:
There we go.